packages/components/modal/src/drawer.tsx

```diff
@@ -167,9 +173,58 @@ export const DrawerContent = forwardRef<DrawerContentProps, "div", false>(
       ...styles.container,
     }
 
+    const getDragDirectionRestriction = () => {
+      switch (placement) {
+        case "top":
+          return { bottom: 0 }
+        case "bottom":
+          return { top: 0 }
+        case "left":
+          return { right: 0 }
+        case "right":
+          return { left: 0 }
+      }
+    }
+
+    const getDragDirection = () => {
+      switch (placement) {
+        case "top":
+          return "y"
+        case "bottom":
+          return "y"
+        case "left":
+          return "x"
+        case "right":
+          return "x"
+      }
+    }
+
+    const isCloseByDragInfo = (info: PanEventInfo) => {
+      switch (placement) {
+        case "top":
+          return info.velocity.y <= -100 || info.offset.y <= -80
+        case "bottom":
+          return info.velocity.y >= 100 || info.offset.y >= 80
+        case "left":
+          return info.velocity.x <= -100 || info.offset.x <= -80
+        case "right":
+          return info.velocity.x >= 100 || info.offset.x >= 80
+      }
+    }
```

Comment:
Please use `useCallback`.

Reply:
I see. I'll try it.

Reply:
Is it right?
```javascript
    const getDragDirectionRestriction = useCallback(() => {
      switch (placement) {
        case "top":
          return { bottom: 0 }
        case "bottom":
          return { top: 0 }
        case "left":
          return { right: 0 }
        case "right":
          return { left: 0 }
      }
    }, [placement]);

    const getDragDirection = useCallback(() => {
      switch (placement) {
        case "top":
        case "bottom":
          return "y"
        case "left":
        case "right":
          return "x"
      }
    }, [placement]);

    const isCloseByDragInfo = useCallback((info: PanEventInfo) => {
      switch (placement) {
        case "top":
          return info.velocity.y <= -100 || info.offset.y <= -80
        case "bottom":
          return info.velocity.y >= 100 || info.offset.y >= 80
        case "left":
          return info.velocity.x <= -100 || info.offset.x <= -80
        case "right":
          return info.velocity.x >= 100 || info.offset.x >= 80
      }
    }, [placement]);
```

Reply:
Yes, if closeOnDrag still work properly with this, this would be correct.